src/components/GoalsComponents/GoalConfigModal/ArchivedAutoComplete.tsx

```diff
@@ -0,0 +1,34 @@
+import React from "react";
+import { GoalItem } from "@src/models/GoalItem";
+import useArchivedGoals from "@src/hooks/useArchivedGoals";
+import AutocompleteComponent from "@src/common/AutoComplete";
+
+interface ArchivedAutoCompleteProps {
+  onGoalSelect: (goal: GoalItem) => void;
+  onInputChange: (value: string) => void;
+  inputValue: string;
+  placeholder: string;
+}
+
+const ArchivedAutoComplete: React.FC<ArchivedAutoCompleteProps> = ({
+  onGoalSelect,
+  onInputChange,
+  inputValue,
+  placeholder,
+}) => {
+  const archivedGoals = useArchivedGoals();
```

Comment:
please use the dexie api for performing the search instead of loading the entire data into the memory